test/webaudio/audioworklet_emscripten_locks.c

```diff
@@ -114,10 +115,16 @@ EM_JS(void, InitHtmlUi, (EMSCRIPTEN_WEBAUDIO_T audioContext), {
 });
 
 bool MainLoop(double time, void* data) {
+  assert(!emscripten_current_thread_is_audio_worklet());
+  static int didUnlock = false;
   switch (whichTest) {
   case TEST_WAIT_ACQUIRE:
-    // Release here to acquire in process
-    emscripten_lock_release(&testLock);
+    if (!didUnlock) {
```

Comment:
Can this be an assertion? IIUC this mode should only ever be reached once?

Reply:
No, it can actually be a hit continuously until the mode changes.   i.e. if the worker is is non responsive then the main thread will just repeatedly call this block forever.    We could probably improve the state machine but I'm not sure I want to start messing with this.

Reply:
Hmm, can it fail to acquire more than once? I guess maybe...